File: meeting-qt/meeting-ipc/nem_hosting_module/service/feedback_service.cpp
/**
 * @copyright Copyright (c) 2021 NetEase, Inc. All rights reserved.
 *            Use of this source code is governed by a MIT license that can be found in the LICENSE file.
 */

#include "nem_hosting_module/service/feedback_service.h"
#include "nem_hosting_module_protocol/protocol/feedback_protocol.h"

NNEM_SDK_HOSTING_MODULE_BEGIN_DECLS
USING_NS_NNEM_SDK_HOSTING_MODULE_PROTOCOL


NEFeedbackServiceIMP::NEFeedbackServiceIMP()
    : IService(ServiceID::SID_Feedback)
    , feedback_srv_listener_(nullptr)
{

}
NEFeedbackServiceIMP::~NEFeedbackServiceIMP()
{

}


void NEFeedbackServiceIMP::feedback(const int& type, const std::string& path, const NEFeedbackCallback& cb)
{
    PostTaskToProcThread(ToWeakCallback([this, type, path, cb]() {
        FeedbackRequest request;
        request.path_ = path;
        request.type_ = type;
        SendData(FeedbackCID::FeedbackCID_feedback, request, IPCAsyncResponseCallback(cb));
    }));
}

void NEFeedbackServiceIMP::addListener(FeedbackServiceListener* listener)
{
    feedback_srv_listener_ = listener;
}

void NEFeedbackServiceIMP::OnPack(int cid, const std::string& data, const IPCAsyncResponseCallback& cb)
{
    switch (cid)
    {
    case FeedbackCID::FeedbackCID_feedback_CB:
    {
        FeedbackResponse response;
        if (response.Parse(data))
        {
            if (feedback_srv_listener_) {
                feedback_srv_listener_->onFeedbackStatus(response.type_, response.error_code_, response.url_);
            }
        }
    }
    break;
    }
}



NNEM_SDK_HOSTING_MODULE_END_DECLS

File: meeting-flutter/meeting_kit/lib/src/meeting_kit/feedback_service.dart
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

part of meeting_kit;

/// 意见反馈服务提供了意见反馈能力。通过这个服务可以提供意见反馈界面入口和意见反馈接口。可通过 [NEFeedbackService] 获取对应的服务实例。
abstract class NEFeedbackService {
  /// 意见反馈接口
  ///
  /// [feedback] 意见反馈的内容
  ///
  /// 结果回调
  Future<NEResult<void>> feedback(NEFeedback feedback);

  /// 展示意见反馈界面
  ///
  Widget loadFeedbackView();
}
